lib/Basic/Subprocess.cpp

```diff
@@ -713,44 +906,45 @@ void llbuild::basic::spawnProcess(
         ProcessInfo info{ attr.canSafelyInterrupt };
         pgrp.add(std::move(guard), pid, info);
       }
-    }
-  }
+
+    // Close the child ends of the forwarded output and control pipes.
+    controlPipeChildEnd.close();
+    outputPipeChildEnd.close();
```

Comment:
The child ends should be closed under the lock because they are potentially inheritable by other processes if we are to spawn one in a parallel thread.